client/my-sites/patterns/home/index.tsx

```diff
@@ -16,10 +18,28 @@ import ImgLayout from './images/layout.svg';
 import ImgPattern from './images/pattern.svg';
 import ImgResponsive from './images/responsive.svg';
 import ImgStyle from './images/style.svg';
-import type { PatternGalleryFC } from 'calypso/my-sites/patterns/types';
+import type { Pattern, PatternGalleryFC } from 'calypso/my-sites/patterns/types';
 
 import './style.scss';
 
+function filterPatternsByQuery( patterns: Pattern[], searchTerm: string ) {
+	if ( ! searchTerm ) {
+		return [];
+	}
+
+	return patterns.filter( ( pattern ) => {
+		const lowercaseSearch = searchTerm.toLowerCase();
+		const patternCategories = Object.values( pattern.categories ).map(
+			( category ) => category?.title
+		);
+		const fields = [ pattern.title, pattern.description, ...patternCategories ].filter(
+			( x ): x is NonNullable< typeof x > => Boolean( x )
+		);
```

Comment:
The filtering code is a bit hard to grasp, maybe we could add a comment like this to clarify its purpose?
```suggestion
		// Filters out any null and undefined values
		const fields = [ pattern.title, pattern.description, ...patternCategories ].filter(
			( x ): x is NonNullable< typeof x > => Boolean( x )
		);
```

Reply:
The Typescript dance is quite silly, but TS can't infer the correct type for `fields` without it. I'm fine with adding a comment to explain.